### DISCUSSION TITLE
How to fetch data through a junction table using the GraphQL API?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#8737] [Archive]
For example, I have 3 tables：
1. `author`
2. `book`
3. `author_book`

The `author_book` table is a junction table connecting the other two tables, containing columns `author_id` and `book_id`.

Now I would like to use a GraphQL query to fetch an `author`'s `book`s. 
However, the `author` table does not contain a column `book`, so this is not doable.

Could anyone teach me how to achieve that?

I know that I can firstly make a call to GraphQL API to fetch `book_id`s from table `author_book` by `author_id`, then make another GraphQL call to fetch the concrete `book_title` from table `book` by the just obtained `book_id`s, but this would be costly due to the extra API call.

Any help would be appreciated!

**Comment by rijkvanzanten** [comment]
@powerseed If you add a many-to-many field called "books" on the "author" collection, you'll be able to do just that 👍🏻

**Reply by powerseed** [reply]
@rijkvanzanten 
I just gave it a try, but I can only query field `author_book.book_id` in the `author.books` field suggested by you, but not `book.book_title`.
Am I missing any configs?

```
query {
	author(filter: { id: { _in: [39] } }) {
                books {
                        book_id  // only this field is allowed, but not "title".
                }
        }
}
```

**Reply by rijkvanzanten** [reply]
A many to many is nothing more than a one-to-many followed by a many-to-one, so the fields you're interfacing with on this "level" are the fields on `author_book`.

Try using

```graphql
query {
	author(filter: { id: { _in: [39] } }) {
                books {
                        book_id {
                                title
                        }
                }
        }
}
```

**Reply by rijkvanzanten** [reply]
> Could you give me a litte bit of hints about the cause?

I think your initial hunch is spot on:

> Wait... It seems that I forgot some permission..

Does your current user have permissions to all three of those collections?

**Reply by powerseed** [reply]
@rijkvanzanten 
You are right, I was missing permission for table `book`.
Problem solved!
Thank you so much for you help!
